src/components/CreateCampaignFromTemplateDialog.tsx

```diff
@@ -6,39 +7,58 @@ import DialogContentText from "@material-ui/core/DialogContentText";
 import DialogTitle from "@material-ui/core/DialogTitle";
 import TextField from "@material-ui/core/TextField";
 import Autocomplete from "@material-ui/lab/Autocomplete";
-import type { TemplateCampaignFragment } from "@spoke/spoke-codegen";
+import type {
+  CreateCampaignFromTemplateMutation,
+  TemplateCampaignFragment
+} from "@spoke/spoke-codegen";
 import {
   GetAdminCampaignsDocument,
   useCreateCampaignFromTemplateMutation,
   useGetTemplateCampaignsQuery
 } from "@spoke/spoke-codegen";
-import React, { useCallback, useMemo, useState } from "react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
 
 export interface CreateCampaignFromTemplateDialogProps {
   organizationId: string;
   open: boolean;
+  onCreateTemplateCompleted: (
+    data: MutationResult<CreateCampaignFromTemplateMutation>["data"],
+    selectedTemplateTitle?: string
+  ) => void;
   onClose?: () => Promise<void> | void;
   defaultTemplate?: TemplateCampaignFragment;
 }
 
 export const CreateCampaignFromTemplateDialog: React.FC<CreateCampaignFromTemplateDialogProps> = (
   props
 ) => {
+  const defaultTemplate = props.defaultTemplate ?? null;
+  const defaultCopyCount = 1;
   const [
     selectedTemplate,
     setSelectedTemplate
-  ] = useState<TemplateCampaignFragment | null>(props.defaultTemplate ?? null);
-  const [quantity, setQuantity] = useState<number | null>(1);
+  ] = useState<TemplateCampaignFragment | null>(defaultTemplate);
+  const [quantity, setQuantity] = useState<number | null>(defaultCopyCount);
   const { data, error } = useGetTemplateCampaignsQuery({
     variables: { organizationId: props.organizationId }
   });
   const [
     createFromTemplate,
     { loading: working }
   ] = useCreateCampaignFromTemplateMutation({
-    refetchQueries: [GetAdminCampaignsDocument]
+    refetchQueries: [GetAdminCampaignsDocument],
+    onCompleted: (onCompletedData) =>
```

Comment:
Suggestion: Add the state changes from `onClose()` to the completion handler in the parent, then just [call `onCreateTemplateCompleted()` instead of  `onClose()` after the mutation returns](https://github.com/politics-rewired/Spoke/pull/1629/files#diff-42c2da86c4e0d4ca05a788365c3695853e2a9e9651a033932219aac1436ddc78R103)

Reply:
I know this is a suggestion, but I want to engage to make sure I know why I'm making certain decisions.

1. By using `onCompleted`, I don't have to do the error checking/data checking myself in `handleClickCreate`. I can guarantee that `handleCreateTemplateCompleted` will be called when we have data and no errors without needing to recreate that logic. I can see an argument about keeping mutation-related logic within `handleClickCreate` in one place. If that's the argument you have, I can move the callback to handleClickCreate and add the data/error check.

2. If I combine onClose to the completion handler `handleCreateTemplateCompleted`, then the completion handler is no longer a pure function. It would become a fn that does "create template completed" state stuff _and_  have unexpected close behavior. This additional behavior wouldn't be obvious from the fn name alone when you're within the context of the CreateCampaignFromTemplateDialog component, which to me is a sign of a badly designed fn.

Reply:
> I can see an argument about keeping mutation-related logic within handleClickCreate in one place. If that's the argument you have, I can move the callback to handleClickCreate and add the data/error check.

this is what i was thinking - i do (personally) prefer having the logic in one place when it's reasonable to

> It would become a fn that does "create template completed" state stuff and have unexpected close behavior. 

feel free to not dive into this 😆 but would you say similar about the `handleDelete` handler here? https://github.com/politics-rewired/Spoke/blob/main/src/containers/AdminTemplateCampaigns/components/TemplateCampaignRow.tsx#L114

Reply:
Ok! I can move the callback logic into handleClickCreate 👍 

I think handleDelete is less of a problem here because every fn called within `handleDelete` is within the same component. But if you wanted to be super clear `handleDelete` should be renamed to `handleDeleteTemplateButtonClick`. The more precise name shows that it does various things onClick of the delete template button rather than being a fn that should only handle deletion of "something".

Reply:
idk if I'm being clear. Precise naming makes it easier to navigate components and switch out the particulars of implementation when refactoring.

Reply:
yup totally makes sense 👍🏾